client/rpc-api/src/system/mod.rs

```diff
@@ -103,4 +103,9 @@ pub trait SystemApi<Hash, Number> {
 	/// Returns the roles the node is running as.
 	#[rpc(name = "system_nodeRoles", returns = "Vec<NodeRole>")]
 	fn system_node_roles(&self) -> Receiver<Vec<NodeRole>>;
+
+	/// Returns the json-serialized chainspec running the node.
+	#[rpc(name = "system_genChainSpec", returns = "jsonrpc_core::Value")]
```

Comment:
```suggestion
	#[rpc(name = "system_getChainSpec", returns = "jsonrpc_core::Value")]
```

?

Reply:
I thought that 'generate' seemed more correct, as we're generating the `HardcodedSync` bit at least.

Reply:
Spelling out the word `generate` may be an improvement here, pending Basti's response.

Reply:
Hmm wait, why is this even a RPC call? I assume this is something that we would call on every release to get a new chain spec with the latest light sync state? I don't see this as a RPC. Why isn't that a CLI command?

Reply:
I like the RPC call and I think that's it's nice to have it, but I do think that a CLI command would be more useful for this purpose. Something that starts a light client, waits until it syncs then outputs a chain spec. Unfortunately, for this to be useful #6880 would need to be solved first.

Reply:
Hmm, I still don't see the value of the RPC call. I first thought that this would be called by a light client, but than I realized that this only works reasonable on light clients. What is the value of the RPC call?

Not sure why #6880 needs to be solved first for this issue? We could just use 2 different base paths.

Another point, why don't we generate the required data for the light sync state on a full node as well?

Reply:
>Another point, why don't we generate the required data for the light sync state on a full node as well?

@cheme @andresilva @svyatonik @rphmeier Is this possible/practical?

Reply:
It is possible, but you'd need to build approximately `(BEST_FINALIZED - 4096) / 2048` tries. From practical point of view, since this RPC call is quite specific (iiuc), it may be implemented. The other option would be to build CHTs on full nodes as well (ofc without pruning headers). This will slow down finalization a bit, though.

Reply:
So it sounds like generating this data on the full node will either take up storage space or cpu time. I think it would be better to just do this on the light client for now.

Reply:
@expenses when do you expect this RPC to be called? And who calls this RPC endpoint?

Reply:
I've gotten rid of the RPC call now.

Reply:
> So it sounds like generating this data on the full node will either take up storage space or cpu time. I think it would be better to just do this on the light client for now.

Can you expand on what you mean here? We want to generate the data on a full node, and use it on light nodes

Reply:
@rphmeier but what's the benefit in that over just gathering it on a light node?

Reply:
I think the idea is that you would be able to generate the data from a node who is already populated (typically a full node) and then give to a machine that wants to run a light node from those data (or at least a subset of it).